### DISCUSSION TITLE
A shortcut to generate image tiles

### DISCUSSION THREAD

**Discussion by joesong168** [libvips/libvips#2728] [Q&A]
Hello @jcupitt ,
   Thanks again for this fantastic tool. One improvement I would like to suggest is that whether we can add a shortcut while generating image tiles by dzsave. As we known, image format like pyramid tiff already has large image tiled in the file itself. Could we just extract these tiles to local filesystem as indicated? Without image format converting, my guess is that this kind of transform should be very fast and efficient.

Regards,
Joe

**Comment by jcupitt** [comment]
Hi @joesong168, 

`dzsave` has this feature now. Try:

```
$ vips dzsave k2.jpg x --depth one --suffix .ppm
$ ls x_files/0/
0_0.ppm  0_6.ppm  1_3.ppm  2_0.ppm  2_6.ppm  3_3.ppm  4_0.ppm  4_6.ppm  5_3.ppm
0_1.ppm  0_7.ppm  1_4.ppm  2_1.ppm  2_7.ppm  3_4.ppm  4_1.ppm  4_7.ppm  5_4.ppm
0_2.ppm  0_8.ppm  1_5.ppm  2_2.ppm  2_8.ppm  3_5.ppm  4_2.ppm  4_8.ppm  5_5.ppm
0_3.ppm  1_0.ppm  1_6.ppm  2_3.ppm  3_0.ppm  3_6.ppm  4_3.ppm  5_0.ppm  5_6.ppm
0_4.ppm  1_1.ppm  1_7.ppm  2_4.ppm  3_1.ppm  3_7.ppm  4_4.ppm  5_1.ppm  5_7.ppm
0_5.ppm  1_2.ppm  1_8.ppm  2_5.ppm  3_2.ppm  3_8.ppm  4_5.ppm  5_2.ppm  5_8.ppm
```

`--depth one` means only generate one level of tiles (not the whole pyramid). `--suffix` lets you set the save format, so in this case just `ppm`, which is a simple binary dump of the values.

**Reply by joesong168** [reply]
Great to hear that. Could you please show my how to extra tiles in a pyramid tiff file encoded in webp and dzsave to IIIF format?

**Reply by jcupitt** [reply]
Sorry, I don't understand. Do you want a pyramid with tiles encoded as webp images?

**Reply by joesong168** [reply]
My understanding of pyramid image is that there are lots of small tiles bundled together inside one large file like some kind of zip file. I am looking for a way to extract these tiles and rearrange them into IIIF format.

**Reply by jcupitt** [reply]
Ah OK, try:

```
vips dzsave my-huge-tiff-file.tif /some/directory/name --layout iiif
```

It'll write an IIIF pyramid in `/some/directory/name`.
